src/plans/D3MOperatorPlan.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+/**
+ *  @title D3M Operator Plan
+ *  @notice An operator sets the desired target assets.
+ */
+contract D3MOperatorPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    uint256                      public enabled;
+
+    address public operator;
+    uint256 public targetAssets;
```

Comment:
Because it is public, this value can be read in place of `getTargetAssets` which would be interpreted incorrectly in case `enabled == 0` (because it seems it is expected to be interpreted as `0`, per `getTargetAssets`'s implementation)

Reply:
Actually, I think that it's better to enforce that the target assets is set back to zero before disabling the Plan. wdyt?

Reply:
I like this because it avoids side-effects which ultimately avoid unexpected/undesired behaviors
Is it compatible with the usual D3M life cycle though?

Reply:
cc @hexonaut

Reply:
We need to be able to disable without requiring target assets is 0 in the case the multisig goes rogue.

Reply:
but any account that's capable of disabling the plan is also capable of setting themselves as operator to set the target assets at zero right?

anyway the current version is also fine to me, but it has the little tradeoff that @Rubilmax is mentioning

Reply:
Or disabling the plan could also set back to zero target assets

Reply:
> but any account that's capable of disabling the plan is also capable of setting themselves as operator to set the target assets at zero right?

Not in the way we have this constructed. This contract can trigger without a delay: https://github.com/makerdao/dss-direct-deposit/blob/master/src/D3MMom.sol#L76

Reply:
I'd just make `targetAssets` a `private` variable.

Reply:
Going to leave this for now unless you guys feel strongly. Code is in audit, and this is somewhat of a superficial change on a backend contract which can be upgraded later if desired.